internal/datacoord/task_index.go

```diff
@@ -207,11 +207,15 @@ func (it *indexBuildTask) PreCheck(ctx context.Context, dependency *taskSchedule
 		}
 	}
 
-	dim, err := storage.GetDimFromParams(field.GetTypeParams())
-	if err != nil {
-		log.Ctx(ctx).Warn("failed to get dim from field type params",
-			zap.String("field type", field.GetDataType().String()), zap.Error(err))
-		// don't return, maybe field is scalar field or sparseFloatVector
+	// Extract dim only for vector types to avoid unnecessary warnings
+	dim := -1
+	if typeutil.IsVectorType(field.GetDataType()) {
```

Comment:
SparseFloatVector doesn't have a `dim` parameter either. It is recommended to use `typeutil.IsFixDimVectorType(field.GetDataType)`.